status-im/status-desktop
# Improve performance by only updating the properties that changed in the model instead of reseting

### What does the PR do

Iterates https://github.com/status-im/status-desktop/pull/16384

Based on top of https://github.com/status-im/status-desktop/pull/16384

We had a lot of cases where we went the lazy way to update an item that changes slightly, where we would reset the whole item, instead of updating the properties that were really updated.

This PR fixes that. The performance of communities should improve a lot, because each time you trigger a dataChanged or reset, it creates a cascade of signals from the QML side, and that could trigger freezes, especially for admins.

### Affected areas

Communities, especially for admins, but for normal users too. Normal chat sections a little too by proxy

### Architecture compliance

My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

In the video, I show that community updates work as intended, as well as member updates and requests.
Ignore the crash of the app on the right, that's a separate bug I'll be opening, but the app on the right is on master, so it's not related to my changes.
There is also the bug where the buttons in the members list opens the profile, that's also on master.

[community-update.webm](https://github.com/user-attachments/assets/28e916dc-daf4-4fe8-b2d5-a28df9dffff1)

### Impact on end user

The app should feel way snappier. There would hopefully never be a freeze when the community gets updated.

### How to test

- Test all sorts of community updates to see that they still apply.
- Test as an admin, owner and member

### Risk 

Tick **one**:

Worst case is that some updated fields don't show as updated until a restart
